### PR TITLE
Add endpoint descriptions to swagger and change reminders endpoints

### PR DESCRIPTION
**Pull request by Daxin18**
Breaking changes:
- GET `api/v1/reminders/{reminderId}/intakes` --> `api/v1/intakes/reminder/{reminderId}`
- POST `api/v1/reminders/{reminderId}/intakes` --> `api/v1/intakes/reminder/{reminderId}`
- GET `api/v1/reminders/intakes/{intakeId}` --> `api/v1/intakes/{intakeId}`
- GET `api/v1/reminders/senior/{seniorId}/intakes` --> `api/v1/intakes/senior/{seniorId}`
- GET `api/v1/medications` --> `api/v1/reminders/medications`

### CODE REVIEW

File: SensoBackend/WebApi/ServiceExtensions.cs [hunk 1]
```diff
@@ -20,7 +21,11 @@ public static void AddWebApiLayer(this IServiceCollection services)
         services.ConfigureOptions<ApiVersioningOptionsSetup>();
         services.ConfigureOptions<ApiExplorerOptionsSetup>();
 
-        services.AddSwaggerGen();
+        services.AddSwaggerGen(options =>
+        {
+            var xmlFilename = $"{Assembly.GetExecutingAssembly().GetName().Name}.xml";
+            options.IncludeXmlComments(Path.Combine(AppContext.BaseDirectory, xmlFilename));
+        });
```
Comment (jakubzehner) [inline SensoBackend/WebApi/ServiceExtensions.cs:28]: Move it to swaggerGenOptionsSetup

File: SensoBackend/WebApi/Controllers/V1/AccountController.cs [hunk 2]
```diff
@@ -20,6 +20,13 @@ public sealed class AccountController(
     IJwtProvider jwtProvider
 ) : ControllerBase
 {
+    /// <summary>
+    /// Creates a new account
+    /// </summary>
+    /// <param name="dto"> Data needed to create an account </param>
+    /// <response code="204"> Account successfully created </response>
+    /// <response code="400"> If dto validation failed </response>
```
Comment (tchojnacki) [inline SensoBackend/WebApi/Controllers/V1/AccountController.cs:28]:
```suggestion
    /// <response code="400"> If DTO validation failed </response>
```

File: SensoBackend/WebApi/Controllers/V1/ProfilesController.cs [hunk 3]
```diff
@@ -80,11 +110,22 @@ public async Task<IActionResult> GetSeniorCaretakers()
         return Ok(caretakers);
     }
 
+    /// <summary>
+    /// Creates a new caretaker profile for the user
+    /// </summary>
+    /// <param name="dto"> Data needed to pair caretaker with the senior </param>
+    /// <response code="200"> Returnsnewly created caretaker profile </response>
```
Comment (tchojnacki) [inline SensoBackend/WebApi/Controllers/V1/ProfilesController.cs:117]:
```suggestion
    /// <response code="200"> Returns newly created caretaker profile </response>
```

File: SensoBackend/WebApi/Controllers/V1/ProfilesController.cs [hunk 4]
```diff
@@ -101,6 +142,15 @@ public async Task<IActionResult> CreateCaretakerProfile(CreateCaretakerProfileDt
         return Ok(profile);
     }
 
+    /// <summary>
+    /// Deletes given caretaker profile
+    /// </summary>
+    /// <param name="caretakerId"> Id of a caretaker </param>
+    /// <param name="seniorId"> Id of a senior </param>
+    /// <response code="200"> Returnsnewly created caretaker profile </response>
```
Comment (tchojnacki) [inline SensoBackend/WebApi/Controllers/V1/ProfilesController.cs:150]:
```suggestion
    /// <response code="200"> Returns newly created caretaker profile </response>
```

File: SensoBackend/WebApi/Controllers/V1/IntakesController.cs [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+[ApiController]
+[Route("api/v{version:apiVersion}/[controller]")]
+[ApiVersion("1.0")]
+public sealed class IntakesController(IMediator mediator) : ControllerBase
+{
+    /// <summary>
+    /// Returns intakes for a senior with a given idd
+    /// </summary>
+    /// <param name="seniorId"> Id of a senior who's reminders are to be returned </param>
```
Comment (tchojnacki) [inline SensoBackend/WebApi/Controllers/V1/IntakesController.cs:20]:
```suggestion
    /// <param name="seniorId"> Id of a senior whose reminders are to be returned </param>
```

File: SensoBackend/WebApi/Controllers/V1/IntakesController.cs [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+namespace SensoBackend.WebApi.Controllers.V1;
+
+[ApiController]
+[Route("api/v{version:apiVersion}/[controller]")]
+[ApiVersion("1.0")]
+public sealed class IntakesController(IMediator mediator) : ControllerBase
+{
+    /// <summary>
+    /// Returns intakes for a senior with a given idd
```
Comment (tchojnacki) [inline SensoBackend/WebApi/Controllers/V1/IntakesController.cs:18]:
```suggestion
    /// Returns intakes for a senior with a given id
```

File: SensoBackend/WebApi/Controllers/V1/RemindersController.cs [hunk 7]
```diff
@@ -82,41 +88,76 @@ public async Task<IActionResult> DeleteReminderById(int reminderId)
         return NoContent();
     }
 
+    /// <summary>
+    /// Returns reminders for a senior with a given id
+    /// </summary>
+    /// <param name="seniorId"> Id of a senior who's reminders are to be returned </param>
```
Comment (tchojnacki) [inline SensoBackend/WebApi/Controllers/V1/RemindersController.cs:94]:
```suggestion
    /// <param name="seniorId"> Id of a senior whose reminders are to be returned </param>
```

**Review by tchojnacki — COMMENTED** [review]
Left some suggestions fixing typos.

We've used both `id`, `Id` and `ID` in the descriptions. Let's keep it consistent (with whichever one, although if I had to choose I'd probably pick `id`).

**Review by tchojnacki — APPROVED** [review]

**Pull request merged**